sympy/series/tests/test_series.py

```diff
@@ -385,3 +386,17 @@ def test_issue_23432():
 def test_issue_23727():
     res = series(sqrt(1 - x**2), x, 0.1)
     assert res.is_Add == True
+
+
+def test_issue_24266():
+    #type1: exp(f(x))
+    assert (exp(-I*pi*(2*x+1))).series(x,0,3) == -1 + 2*I*pi*x + 2*pi**2*x**2 + O(x**3)
+    assert (exp(-I*pi*(2*x+1))*gamma(1+x)).series(x,0,3) == -1 + x*(EulerGamma + 2*I*pi) + x**2*(-EulerGamma**2/2 + 23*pi**2/12 - 2*EulerGamma*I*pi) + O(x**3)
```

Comment:
I think you could format this better. Take a couple terms on the next line, this feels a bit cumbersome.